### ISSUE TITLE
USC/CTC.Stage2And3.rnaseq.20160323

### ISSUE THREAD

**Issue by melissacline** [BD2KGenomics/core-operations#23]
This is RNASeq data from USC, affiliated with ISPY2 but a separate institution.  The RNASeq-cgl pipeline is to be run on it.  It should get done, but there's no tip-top priority urgency.  See the file USC/CTC.Stage2And3.rnaseq.20160323.23.input.txt  See also the github issue https://github.com/BD2KGenomics/driver-projects/issues/44#issuecomment-200571116

**Comment by Jeltje** [comment]
@jvivian @melissacline The logical output location for these files would be
cgl-driver-projects/usc/rnaseq_cgl_20160328
However, there are already a lot of tar.gz files directly under cgl-driver-projects/usc/, notably with no indication of what pipeline was run on those.

Should I create my subdirectory and proceed? Or do I simply add the output files directly to cgl-driver-projects/usc/ ?

**Comment by Jeltje** [comment]
On a related note, does s3am make it possible to rename our own files? That way we can create a separate subdirectory for the existing files and move them (I know they're not truly subdirectories but let's pretend). It does mean I'd need to update the corresponding git issue but that should be OK.

**Comment by melissacline** [comment]
@Jeltje , do you have everything you need to get started?

**Comment by Jeltje** [comment]
@melissacline @jvivian I can get started on this current bunch but I'd still like some info on the previous run. The files at cgl-driver-projects/usc/ (which is where I want to put my output) do not seem to have an issue associated with them. Which pipeline was run on those (names are like Sample_36978.tar.gz)? Do I add my output to this directory? Can we delete the files that are there?
I'm just trying to prevent this from becoming even more disorganized.

**Comment by jvivian** [comment]
They were run with the CGL rna-seq pipeline and were the Stage 4 outputs
from USC.  You can re-S3AM them to cgl-driver-projects/usc/stage4 then
delete them to better organize them if you'd like.

**Comment by Jeltje** [comment]
Thanks! @melissa, do we need to keep these files around?

**Comment by melissacline** [comment]
Let's keep the output files for now (they should be small).  We don't need
to keep the input files.  As I remember, those inputs showed a lot of
contamination anyway...

**Comment by Jeltje** [comment]
Reorganized all input and output files for USC stage 4 and stage 2-3.
Started rnaseq_cgl run

**Comment by Jeltje** [comment]
11 out of 13 files finished successfully. JL09182014.tar.gz had a mismatched reads error and Sample_78956CTC wasn't run for reasons I can't discern. I will look into JL09182014 and rerun the other sample

**Comment by melissacline** [comment]
Thanks!

**Comment by Jeltje** [comment]
JL09182014.tar.gz unpacks into a bunch of subdirectories with names like Sample_CTC-36260 and Sample_CTC-37232. If these are truly different samples, we're running this wrong and they should be submitted independently.
The sample directories appear to contain individual fastq files as well as concatenated ones:
CTC-36260_AACCAG_L001_R1_001.fastq.gz
CTC-36260_AACCAG_L001_R2_001.fastq.gz
CTC-36260_AACCAG_L002_R1_001.fastq.gz
CTC-36260_AACCAG_L002_R1_003.fastq.gz
CTC-36260_AACCAG_L002_R2_001.fastq.gz
CTC-36260_AACCAG_L002_R2_003.fastq.gz
CTC-36260_R1_merged.fastq.gz
CTC-36260_R2_merged.fastq.gz
Indeed the CTC_R1_merged.fastq.gz contains the other _R1_ files. Our code will grab both of them.

This is another strong argument for not submitting tar files.

**Comment by Jeltje** [comment]
Other samples in the submission fortunately look normal (no merged files, no mixed samples). The samples inside JL09182014.tar.gz do have similar names to submitted samples, but are not identical. 

@melissacline can you talk to the submitter about this? They may want to submit JL09182014's subdirectories as separate samples.

**Comment by Jeltje** [comment]
Started rerun of Sample_78956CTC

**Comment by melissacline** [comment]
I will.  Let me make sure I understand the issues.
1. Instead of getting a tarball per-sample that contains two fastqs, we've
received a tarball with data for many samples.  i.e. the packaging is not
what we need.
2. The tarballs contain what appears to be merged files as well as
individual files (perhaps the files that had been merged).  For our
pipeline, this is extraneous (if we have the merged files, we don't need
the individual files) and harmful (the code will grab all fastqs from the
tarball, so if the individual files represent duplicated data, it will grab
that data twice).
Is that correct?

**Comment by Jeltje** [comment]
Almost. I actually checked that the merged file contains the individual files, and it does.
1. Instead of getting a tarball per-sample that contains _R1 and R2 fastqs_ , we've received a tarball _(JL09182014.tar.gz)_ with data for _what looks like several_ samples. i.e. the packaging is not what we need. 
2. _This JL09182014.tar.gz_ tarball (but not the others) contains merged files as well as individual files (i.e. the files that had been merged). For our pipeline, this is extraneous (if we have the merged files, we don't need the individual files) and harmful (the code will grab all fastqs from the tarball, so if the individual files represent duplicated data, it will grab that data twice).

And while we're at it:
3. Sample_78956CTC.tar.gz contains a fastq pair that's incomplete:
78956CTC_AGCATG_L007_R1_016.fastq.gz has more reads than its R2 companion. This breaks our pipeline. I also get a gzip error ("unexpected end of file") so it looks like the R2 file wasn't copied properly at some point.
